src/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -2607,6 +2608,33 @@ public static void parseTownClaimCommand(Player player, String[] split) {
 				TownyMessaging.sendDebugMsg("townClaim: Post-Filter Selection " + Arrays.toString(selection.toArray(new WorldCoord[0])));
 				checkIfSelectionIsValid(town, selection, attachedToEdge, blockCost, false);
 
+				//Check if other plugins have a problem with claiming this area
+				int blockedClaims = 0;
+
+				for(WorldCoord coord : selection){
+					TownyWorld coordWorld;
+					TownPreClaimEvent preClaimEvent;
+					try {
+						coordWorld = coord.getTownyWorld();
+						preClaimEvent = new TownPreClaimEvent(new TownBlock(coord.getX(), coord.getZ(), coordWorld));

```

Comment:
Pass the town alongside the townblock, this will make it so the TownBlock object doesn't need changing. I'm worried about the way it is not causing the townblock to record the town entry, even if the town doesn't add it on it's end. A townblock that thinks its part of a town tends to render that piece of wilderness unclaimable and it's tedious to fix manually if Towny doesn't auto clean it up.